c7n/resources/quotas.py

```diff
@@ -154,19 +157,31 @@ def process(self, resources, event):
         start_time = end_time - timedelta(1)
 
         limit = self.data.get('limit', 80)
+        min_period = max(self.data.get('min_period', 300), 60)
 
         result = []
 
         for r in resources:
             metric = r.get('UsageMetric')
-            if not metric:
+            quota = r.get('Value')
+            if not metric or quota is None:
                 continue
             stat = metric.get('MetricStatisticRecommendation', 'Maximum')
             if stat not in self.metric_map and self.percentile_regex.match(stat) is None:
                 continue
+
+            period_value = {"PeriodValue": 1, "PeriodUnit": "SECOND"}
+            # NOTE Hot fix for "QuotaName": "Concurrently executing Automations"
+            if r.get("QuotaCode") == "L-09101E66" and "Period" not in r:
```

Comment:
what is the rationale for this specific quota?

Reply:
Last time I check, the quota does not have a period attribute. Without that the new added scale logic won't work.

Reply:
Looking into this it seems like there's actually quite a few quotas that have this behavior, I was trying to figure out a more generic solution for this but I was distracted with the docker ci work, i'll try to take a look at this again this week

Reply:
@kentnsw I agree with @thisisshi comment. I don't understand why we want to specifically check for that quota. Isn't it enough to keep the existing code where we default period to `int(timedelta(1).total_seconds())` if Period key doesn't exist? And then, check if period is less than min_period. If it is, then do the scaling to avoid the limit issue. The code would look something like this
```
            if 'Period' in r:
                period_unit = self.time_delta_map[r['Period']['PeriodUnit']]
                period = int(timedelta(**{period_unit: r['Period']['PeriodValue']}).total_seconds())
            else:
                period = int(timedelta(1).total_seconds())

           # Use scaling to avoid CW limit of 1440 data points
            metric_scale = 1
            if period < min_period and stat == "Sum":
                metric_scale = min_period / period
                period = min_period
```

Reply:
@darrendao cool! why it isn't me who come up that idea :D
Will change it later